flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/reuse/ScanReuserUtils.java

```diff
@@ -329,6 +329,12 @@ public static String getDigest(CommonPhysicalTableSourceScan scan, boolean witho
         TableSourceTable table = scan.tableSourceTable();
         List<String> digest = new ArrayList<>();
         digest.addAll(table.getNames());
+
+        // input should be the first item
+        if (!scan.getInputs().isEmpty()) {
+            digest.add("input=[" + scan.getInputs() + "]");
```

Comment:
The length of `scan.getInputs()`  will be very long? I think you can simplify the output of the digest here